client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -341,9 +343,14 @@ private void throwIfClosed() throws TxnFailedException {
                 RuntimeException::new);
         UUID txnId = txnSegments.getTxnId();
         Map<Segment, SegmentTransaction<Type>> transactions = new HashMap<>();
+        DelegationTokenProvider tokenProvider = null;
         for (Segment s : txnSegments.getSteamSegments().getSegments()) {
-            SegmentOutputStream out = outputStreamFactory.createOutputStreamForTransaction(s, txnId,
-                    config, txnSegments.getSteamSegments().getDelegationToken());
+            if (tokenProvider == null) {
+                // We can share token providers across segments as delegation tokens are scope + stream specific.
+                tokenProvider = DelegationTokenProviderFactory.create(txnSegments.getSteamSegments().getDelegationToken(),
```

Comment:
nit: typo `getSteamSegment()` to `getStreamSegments()`

Reply:
Sure. This was an existing code with a typo. I've changed it to `getStreamSegments()` in the last commit. You sure have a keen eye. :)